tailscale/resource_aws_external_id.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) David Bond, Tailscale Inc, & Contributors
+// SPDX-License-Identifier: MIT
+
+package tailscale
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	tsclient "github.com/tailscale/tailscale-client-go/v2"
+)
+
+func resourceAWSExternalID() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The aws_external_id resource allows you to mint an AWS External ID that Tailscale can use to assume an AWS IAM role that you create for the purposes of allowing Tailscale to stream logs to your S3 bucket. See the logstream_configuration resource for more.",
+		CreateContext: resourceAWSExternalIDCreate,
+
+		// No GET or DELETE endpoints in the API. This is a create-only resource.
+		ReadContext:   schema.NoopContext,
+		DeleteContext: schema.NoopContext,
+
+		Schema: map[string]*schema.Schema{
+			"external_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The actual External ID that Tailscale will supply when assuming your role. You probably want to reference this in your IAM role's trust policy.",
+			},
+			"tailscale_aws_account_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The AWS account from which Tailscale will assume your role. You probably want to reference this in your IAM role's trust policy.",
+			},
+		},
+	}
+}
+
+func resourceAWSExternalIDCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*tsclient.Client)
+
+	aid, err := client.Logging().CreateOrGetAwsExternalId(ctx, tsclient.CreateOrGetAwsExternalIdRequest{Reusable: false})
```

Comment:
It might be worth a comment on why "reusable" is false, and why it must be.

Reply:
@oxtoacart I pushed up [a change](https://github.com/tailscale/terraform-provider-tailscale/compare/21d8c90f1a7dc5bb0a3392391ef29f518266c72d..6415cdae90931e7b5febd0ef2ca627362da32ba2#diff-4323850be6e4e1e73caf830d8915901a6ba80d9a2318c892af3fe5b2e0f29e2cR42) that includes a (perhaps overlong) comment about this. Take a look and let me know if you think it's helpful.

But it's possible that I'm missing the point here, because I actually don't see why it _must_ be: the reasons I see are that there's no real need, and in certain circumstances (not that common, though) it could lead to errors. 

If I'm missing the point and there's a much simpler explanation please let me know.

Reply:
I think(?) the explanation is something like this:

We have to set `reusable` to `false`. Otherwise, multiple external ID resources could end up with the same Terraform ID.

Side note - I don't fully understand the implications of two different resources having the same ID, but I suspect it's not good.

Reply:
Updated!

Reply:
And thanks for explaining. I think you're right about that. Resource IDs are supposed to be unique and Terraform would be unhappy.